Sources/Identity/CustomerInfoManager.swift

```diff
@@ -181,13 +181,18 @@ class CustomerInfoManager {
     }
 
     func cache(customerInfo: CustomerInfo, appUserID: String) {
-        do {
-            let jsonData = try JSONEncoder.default.encode(customerInfo)
-            self.withData { $0.deviceCache.cache(customerInfo: jsonData, appUserID: appUserID) }
-            self.sendUpdateIfChanged(customerInfo: customerInfo)
-        } catch {
-            Logger.error(Strings.customerInfo.error_encoding_customerinfo(error))
+        if customerInfo.shouldCache {
+            do {
+                let jsonData = try JSONEncoder.default.encode(customerInfo)
+                self.withData { $0.deviceCache.cache(customerInfo: jsonData, appUserID: appUserID) }
+            } catch {
+                Logger.error(Strings.customerInfo.error_encoding_customerinfo(error))
+            }
+        } else {
+            Logger.debug(Strings.customerInfo.not_caching_offline_customer_info)
         }
+
+        self.sendUpdateIfChanged(customerInfo: customerInfo)
```

Comment:
This will also send the updated customer info in case it goes in the catch.  We were not doing that before, and I am not sure we want to?

Reply:
Oh yeah, I meant to write a unit test for that.
I think it's still useful to notify the user, even if it fails to encode. I know that's changing behavior, but it made this easier so we didn't need 3 branches.

Reply:
Oh there's no easy way to test it cause we can't really represent a `CustomerInfo` that would fail to encode :/

Reply:
I guess it's fine, the reason it won't be able to be encoded is not that the customer info is broken, so I think it's safe to send it anyways

Reply:
Agreed.